src/core/lib/iomgr/call_combiner.cc

```diff
@@ -84,7 +75,7 @@ void grpc_call_combiner_start(grpc_exec_ctx* exec_ctx,
       gpr_log(GPR_DEBUG, "  EXECUTING IMMEDIATELY");
     }
     // Queue was empty, so execute this closure immediately.
-    GRPC_CLOSURE_SCHED(exec_ctx, closure, error);
+    run_now = true;
```

Comment:
Note that this only takes effect if the new closure is going to run immediately.  If it gets queued and then runs later, it will be invoked with `GRPC_CLOSURE_SCHED()` instead of `GRPC_CLOSURE_RUN()`, regardless of whether it was added via `GRPC_CALL_COMBINER_SCHED()` or `GRPC_CALL_COMBINER_RUN()` (see line 160 below).  In particular, this means that anywhere that we call `GRPC_CALL_COMBINER_RUN()` while we're already holding the call combiner (which is the case in most of the filter code), this PR is a no-op.

Taking a step back, what is the goal of this PR?  It's not obvious to me why it matters whether we use `GRPC_CLOSURE_SCHED()` or `GRPC_CLOSURE_RUN()` for call combiner calls.